gcovr/gcov.py

```diff
@@ -575,12 +575,18 @@ def apply_filter_include_exclude(filename, include_filters, exclude_filters):
         excluded (bool): True when filename failed the exclude_filters
     """
 
-    filtered = not any(f.match(filename) for f in include_filters)
+    filtered = not any(
+        f.match(filename) or f.match(realpath(filename))
```

Comment:
If the filters need a realpath() to work, the filters should do this themselves. Most filters already do this correctly, only DirectoryPrefixFilter needs this.

Reply:
I'll check this.

Reply:
Ok, it's working.

Reply:
Test was green because of `skip` instead of `skipif`. Now the test is failing.
The `DirectoryPrefixFilter` fails for the realpath here. Thats the reason wy abspath and realpath was used.

Reply:
Now we check in `DirectoryPrefixFilter` the abspath and the realpath. This is needed because realpath is used for the filter directory and if the filter directory is a link itself the the realpath is also needed for the file.